Joohwan Ko/partner_selector.py

```diff
@@ -0,0 +1,175 @@
+"""
+Module for selecting partners at Initialization Period
+
+@author: Joohwan Ko
+"""
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from itertools import combinations
+from tqdm import tqdm
+from statsmodels.distributions.empirical_distribution import ECDF
+
+
+class SelectPartner:
+    """
+    Implementation for section 3.1.1: Partner selection from the paper "Statistical arbitrage with vine copulas(2016,Stübinger)
+    and this handles 4 different ways of selecting partner at Initialization Period
+    1. Traditional Approach
+    2. Extended Approach
+    3. Geometric Approach
+    4. Extremal Approach
+    """
+
+    def __init__(self, rank_df):
+        self.rank_df = rank_df
+        self.rank_corr_df = rank_df.corr()
+
+    def get_partner(self, method='traditional', extended_rho='first'):
+        """
+        Select partners for copula by choosing a method
+        :param method: (str) {'traditional', 'extended', 'geometric', 'extremal'}, default: 'traditional'
+        :param extended_rho: (str) {'first', 'second', 'third'}, default: 'first'
+        :return: (list) Output matrix quadruple Q
+        """
+        Q_matrix = []
+        total_stocks = self.rank_df.columns
+        for ticker in tqdm(total_stocks):
+            # Set a variable which indicates sum of the highest pairwise correlations
+            highest_sum = float('-inf')
+            # Make a list of the best quadruple for a given target stock
+            best_quadruple = []
+            # Get a list without the target stock
+            other_stocks = [x for x in total_stocks if x != ticker]
```

Comment:
Add more white spaces. Your code is very clumped together and readability is suffering.